cmd/snap/main.go

```diff
@@ -68,12 +70,14 @@ var experimentalCommands []*cmdInfo
 
 // addCommand replaces parser.addCommand() in a way that is compatible with
 // re-constructing a pristine parser.
-func addCommand(name, shortHelp, longHelp string, builder func() flags.Commander) *cmdInfo {
+func addCommand(name, shortHelp, longHelp string, builder func() flags.Commander, optDescs map[string]string, argDescs [][2]string) *cmdInfo {
```

Comment:
Maybe its just me, but I am not fond of `argDescs [][2]string`. Maybe `argDescs []Arg` with `struct Arg {Name, Descr string} ` Slightly more verbose but easier to read (I hope). WDYT?